# Author - Mayank Singh
Data Analyst|MySQL|Python|Web Crawling and Scraping
|EDA|Visualization|CBAP|Agile Cert

Creation Date - 09-10-2022


# Importing Liabrary

In [1]:
import pandas as pd

# Reading File

In [2]:
file = pd.read_csv("C:\\Users\\mayank.singh27\\PycharmProjects\\pythonProject\\TestCode\\AAP_Donation_Data_fy20-21.csv", encoding_errors= 'replace')

In [3]:
file.head()

Name  \
0  1.? Bharatha Swamukti Samsthe   
1                            NaN   
2    2.? Prudent Electoral Trust   
3                            NaN   
4  3.? Bharatha Swamukti Samsthe   

                                             Address  Pan  \
0  108 Sarang of Unitech Parkway Apts 1st Main Ro...    Y   
1                                                NaN  NaN   
2  G-15 Hans Bhawan 1 Bahadur Shah Zafar Marg New...    Y   
3                                                NaN  NaN   
4  108 Sarang of Unitech Parkway Apts 1st Main Ro...    Y   

                      Amount ?  \
0  Rs. 1,20,00,000.00 �1�Cror+   
1                          NaN   
2  Rs. 1,00,00,000.00 �1�Cror+   
3                          NaN   
4   Rs. 99,00,000.00 �99�Lacs+   

                                   Contribution Mode Financial  
0  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
1                                                NaN       NaN  
2  Cheque HDFC Bank Kailash Building K.G. Marg Ne...   2020-21  
3                                                NaN       NaN  
4  Cheque State Bank of India Mahalakshmi Layout ...   2020-21

# Cleaning Data

In [4]:
file.dropna(inplace = True) # droping blank/Null value columns
file['Name'] = file['Name'].str.replace('\d+','') # replacing digits form Name Column
file['Name'] = file['Name'].str.replace('.','') # replacing . from Name column
file['Name'] = file['Name'].str.replace('?','') # replacing special char ? from Name Column
file['Amount ?'] = file['Amount ?'].str.replace('�','') # replacing special char � from Name Column
file['Amount ?'] = file['Amount ?'].str.replace('Rs','')  # replacing Rs from Amount column
file['Amount ?'] = file['Amount ?'].str.replace(',','')
file['Amount ?'] = file['Amount ?'].str.lstrip('.') # replacing . from Amount column
file['Amount ?'] = file['Amount ?'].str.rstrip('+') # replacing + from Amount column
file['Amount ?'] = file['Amount ?'].str.rstrip('cror') # replacing cror from Amount column
file['Amount ?'] = file['Amount ?'].str.rstrip('C') # replacing C from Amount column
file['Amount ?'] = file['Amount ?'].str.rstrip('Lacs') # replacing Lacs from Amount column
file['Amount ?'] = file['Amount ?'].str.split().str[0] # replacing unwanted digits after space from Amount column

C:\Users\mayank.singh27\AppData\Local\Temp\ipykernel_29944\2697003470.py:2: FutureWarning: The default value of regex will change from True to False in a future version.
  file['Name'] = file['Name'].str.replace('\d+','') # replacing digits form Name Column
C:\Users\mayank.singh27\AppData\Local\Temp\ipykernel_29944\2697003470.py:3: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  file['Name'] = file['Name'].str.replace('.','') # replacing . from Name column
C:\Users\mayank.singh27\AppData\Local\Temp\ipykernel_29944\2697003470.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  file['Name'] = file['Name'].str.replace('?','') # replacing special char ? from Name Column


# Cleaned data Set

In [5]:
file.head()

Name  \
0   Bharatha Swamukti Samsthe   
2     Prudent Electoral Trust   
4   Bharatha Swamukti Samsthe   
6   Bharatha Swamukti Samsthe   
8   Bharatha Swamukti Samsthe   

                                             Address Pan     Amount ?  \
0  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y  12000000.00   
2  G-15 Hans Bhawan 1 Bahadur Shah Zafar Marg New...   Y  10000000.00   
4  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9900000.00   
6  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9500000.00   
8  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9100000.00   

                                   Contribution Mode Financial  
0  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
2  Cheque HDFC Bank Kailash Building K.G. Marg Ne...   2020-21  
4  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
6  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
8  Cheque State Bank of India Mahalakshmi Layout ...   2020-21

# Cleaning Headers - 'Amount ?'

In [6]:
file.columns = file.columns.str.replace ('?', '')

C:\Users\mayank.singh27\AppData\Local\Temp\ipykernel_29944\434693188.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  file.columns = file.columns.str.replace ('?', '')


# Removing extra spaces before and after the columns' Data

In [7]:
file['Name'] = file['Name'].str.strip()
file['Address'] = file['Address'].str.strip()
file['Pan'] = file['Pan'].str.strip()
file['Amount '] = file['Amount '].str.strip()
file['Contribution Mode'] = file['Contribution Mode'].str.strip()
file['Financial'] = file['Financial'].str.strip()
file.columns = file.columns.str.strip()

# Final Cleaned Data Set

In [8]:
file

Name  \
0     Bharatha Swamukti Samsthe   
2       Prudent Electoral Trust   
4     Bharatha Swamukti Samsthe   
6     Bharatha Swamukti Samsthe   
8     Bharatha Swamukti Samsthe   
...                         ...   
3438         Ramesh Bellamkonda   
3440         Ramesh Bellamkonda   
3442         Ramesh Bellamkonda   
3444         Ramesh Bellamkonda   
3446         Ramesh Bellamkonda   

                                                Address Pan       Amount  \
0     108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y  12000000.00   
2     G-15 Hans Bhawan 1 Bahadur Shah Zafar Marg New...   Y  10000000.00   
4     108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9900000.00   
6     108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9500000.00   
8     108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9100000.00   
...                                                 ...  ..          ...   
3438  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y        10.00   
3440  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y        10.00   
3442  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y        10.00   
3444  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y        10.00   
3446  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y        10.00   

                                      Contribution Mode Financial  
0     Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
2     Cheque HDFC Bank Kailash Building K.G. Marg Ne...   2020-21  
4     Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
6     Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
8     Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
...                                                 ...       ...  
3438                                               Cash   2020-21  
3440                                               Cash   2020-21  
3442                                               Cash   2020-21  
3444                                               Cash   2020-21  
3446                                               Cash   2020-21  

[1724 rows x 6 columns]

# Saving to New file And Reading New Data Set

In [9]:
# file.to_csv("C:\\Users\\mayank.singh27\\PycharmProjects\\pythonProject\\TestCode\\AAP_Donation_Data_fy20-21_Cleaned_New.csv")

In [10]:
df = pd.read_csv("C:\\Users\\mayank.singh27\\PycharmProjects\\pythonProject\\TestCode\\AAP_Donation_Data_fy20-21_Cleaned_New.csv")

In [11]:
df.head()

Unnamed: 0                       Name  \
0           0  Bharatha Swamukti Samsthe   
1           2    Prudent Electoral Trust   
2           4  Bharatha Swamukti Samsthe   
3           6  Bharatha Swamukti Samsthe   
4           8  Bharatha Swamukti Samsthe   

                                             Address Pan      Amount  \
0  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y  12000000.0   
1  G-15 Hans Bhawan 1 Bahadur Shah Zafar Marg New...   Y  10000000.0   
2  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9900000.0   
3  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9500000.0   
4  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9100000.0   

                                   Contribution Mode Financial  
0  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
1  Cheque HDFC Bank Kailash Building K.G. Marg Ne...   2020-21  
2  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
3  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
4  Cheque State Bank of India Mahalakshmi Layout ...   2020-21

# Getting Started with EDA

In [12]:
# 1. Basic Info about Data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1724 entries, 0 to 1723
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         1724 non-null   int64  
 1   Name               1724 non-null   object 
 2   Address            1724 non-null   object 
 3   Pan                1724 non-null   object 
 4   Amount             1724 non-null   float64
 5   Contribution Mode  1724 non-null   object 
 6   Financial          1724 non-null   object 
dtypes: float64(1), int64(1), object(5)
memory usage: 94.4+ KB


info() provides us basic overview of the data:
1. Total columns = 7
2. There is no - null values i.e. we have values available in every column
3. Data type of each coulmn, one (Amount) is in float, one is in int (Unnamed 0) and all others are object/string
4. Total entries = 1724

In [13]:
## with .info() and .head(), we can see that there is a non useful column --Unnamed 0--

In [14]:
df = df.drop('Unnamed: 0', axis = 1) # droping the non useful column completely

In [15]:
df.head()

Name  \
0  Bharatha Swamukti Samsthe   
1    Prudent Electoral Trust   
2  Bharatha Swamukti Samsthe   
3  Bharatha Swamukti Samsthe   
4  Bharatha Swamukti Samsthe   

                                             Address Pan      Amount  \
0  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y  12000000.0   
1  G-15 Hans Bhawan 1 Bahadur Shah Zafar Marg New...   Y  10000000.0   
2  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9900000.0   
3  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9500000.0   
4  108 Sarang of Unitech Parkway Apts 1st Main Ro...   Y   9100000.0   

                                   Contribution Mode Financial  
0  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
1  Cheque HDFC Bank Kailash Building K.G. Marg Ne...   2020-21  
2  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
3  Cheque State Bank of India Mahalakshmi Layout ...   2020-21  
4  Cheque State Bank of India Mahalakshmi Layout ...   2020-21

In [16]:
Toatl_Donation_Amount = df['Amount'].sum()

In [17]:
Toatl_Donation_Amount

113287398.0

AAP received a total of 11 cror 32 lacs 87 thousand 398 INR Donation in FY - 20-21

# Converting different contribution_mode to uniform Payment_mode using keywords 'Bank, Cheque, Online, Cash'

In [18]:
df['Payment_mode'] = df['Contribution Mode'].str.split().str[0] #step 1 spliting the sentence into individual 
                                                                #word seperated by space and extrcating word at index [0]
                                                                #step 2 Replacing non uniform words with uniform keywords 
                                                                #from sentence     
df['Payment_mode'] = df['Payment_mode'].str.replace('Cheaue', 'Cheque')
df['Payment_mode'] = df['Payment_mode'].str.replace('Cheuqe', 'Cheque')
df['Payment_mode'] = df['Payment_mode'].str.replace('HDFC', 'Bank')
df['Payment_mode'] = df['Payment_mode'].str.replace('J', 'Bank')
df['Payment_mode'] = df['Payment_mode'].str.replace('SVC', 'Bank')
df['Payment_mode'] = df['Payment_mode'].str.replace('State', 'Bank')

In [19]:
Amt_by_diff_modes = df.groupby('Payment_mode').sum()

In [20]:
Amt_by_diff_modes

Amount
Payment_mode            
Bank           2135000.0
Cash               110.0
Cheque        94853272.0
Online        16299016.0

Total amount transferred/contributed by different payment modes. Highest donation was given via ##Cheque i.e. approx = 9.5 crores 

In [21]:
PAN_NoPAN_contri = df.groupby('Pan').sum()

In [22]:
PAN_NoPAN_contri

Amount
Pan             
N       305500.0
Y    112981898.0

Out of Total donation of appro 11.35 crores donation of INR 3,05,500.00 was given without declaring PAN number. 

In [23]:
Persons = df.groupby('Name').sum()

In [24]:
Persons.sort_values(by = 'Amount',ascending=False )

Amount
Name                                         
Bharatha Swamukti Samsthe          51700000.0
Prudent Electoral Trust            17000000.0
Ramesh Bellamkonda                  6400100.0
Nohar Chand & Sons Steels (P) Ltd   5100000.0
Kama Realty (Delhi) Limited         5000000.0
...                                       ...
Dr Ashok Kumar K Lodha                 7000.0
Prabhat Kumar Sharma                   6000.0
Ramanujam Krishnakumar                 5000.0
M Shihabudheen                         1500.0
Shantala                                 10.0

[291 rows x 1 columns]

Highest Donation was given by "Bharatha Swamukti Samsthe" - which is INR - 5,17,00,000, i.e. more than 5 crores
similarly we can see top 3 donators after Bharatha Swamukti Samsthe are:
1. Prudent Electoral Trust
2. Ramesh Bellamkonda
3. Nohar Chand & Sons Steels (P) Ltd	

In [25]:
Person_mode = df.groupby(['Name', 'Payment_mode']).sum()

In [26]:
Person_mode.sort_values(by = 'Amount',ascending=False )

Amount
Name                              Payment_mode            
Bharatha Swamukti Samsthe         Cheque        51700000.0
Prudent Electoral Trust           Cheque        15000000.0
Ramesh Bellamkonda                Cheque         6400000.0
Nohar Chand & Sons Steels (P) Ltd Cheque         5100000.0
Kama Realty (Delhi) Limited       Cheque         5000000.0
...                                                    ...
Prithvi Reddy                     Online            2021.0
M Shihabudheen                    Online            1500.0
Ramesh Bellamkonda                Cash               100.0
Prithvi Reddy                     Cash                10.0
Shantala                          Online              10.0

[311 rows x 1 columns]

All the Top Donators had contributed via Cheque

In [27]:
Person_mode

Amount
Name                              Payment_mode          
Aakash Jain                       Online         44800.0
Aakash Sharma                     Cheque         15000.0
                                  Online          5800.0
Aareen Healthcare Private Limited Cheque        200000.0
Abbas Kaiyum Sutarwala            Online         23580.0
...                                                  ...
Vikram Fernandes                  Online         35000.0
Vinodkumar A Pillai               Online         86000.0
Yogeshbhai Valjibhai Makwana      Cheque         21000.0
                                  Online          6900.0
Zaheer Khan                       Online         26942.0

[311 rows x 1 columns]

There are people who have donated via more than one mode.